### PR TITLE
Mage_Core - IDE improvements

### PR DESCRIPTION
**Pull request by sreichel**
- added/fixed missing DOCs
- add @method annotations
- changed @return to $this for methods and annotations
- changed magic methods on Varien_Objects to setData()/getData()
- flipped @var annotations
- ...

### CODE REVIEW

File: app/code/core/Mage/Core/Helper/Data.php [hunk 1]
```diff
@@ -200,7 +199,7 @@ public function formatTime($time = null, $format = Mage_Core_Model_Locale::FORMA
 
         if (is_null($time)) {
             $date = Mage::app()->getLocale()->date(time());
-        } else if ($time instanceof Zend_Date) {
+        } elseif ($time instanceof Zend_Date) {
```
Comment (kkrieger85) [inline app/code/core/Mage/Core/Helper/Data.php]: Difference `else if` and `elseif`:

> Note: Note that elseif and else if will only be considered exactly the same when using curly brackets as in the above example. When using a colon to define your if/elseif conditions, you must not separate else if into two words, or PHP will fail with a parse error.

#ref https://www.php.net/manual/en/control-structures.elseif.php

File: app/code/core/Mage/Core/Model/App/Emulation.php [hunk 2]
```diff
@@ -88,9 +90,9 @@ public function startEnvironmentEmulation($storeId, $area = Mage_Core_Model_App_
         $initialLocaleCode = $this->_emulateLocale($storeId, $area);
 
         $initialEnvironmentInfo = new Varien_Object();
-        $initialEnvironmentInfo->setInitialTranslateInline($initialTranslateInline)
-            ->setInitialDesign($initialDesign)
-            ->setInitialLocaleCode($initialLocaleCode);
+        $initialEnvironmentInfo->setData('initial_translate_in_line', $initialTranslateInline)
```
Comment (kkrieger85) [inline app/code/core/Mage/Core/Model/App/Emulation.php]: As far as I read this line it should be

`$initialEnvironmentInfo->setData('initial_translate_inline', $initialTranslateInline)`

File: app/code/core/Mage/Core/Model/App/Emulation.php [hunk 3]
```diff
@@ -106,12 +108,12 @@ public function startEnvironmentEmulation($storeId, $area = Mage_Core_Model_App_
      */
     public function stopEnvironmentEmulation(Varien_Object $initialEnvironmentInfo)
     {
-        $this->_restoreInitialInlineTranslation($initialEnvironmentInfo->getInitialTranslateInline());
-        $initialDesign = $initialEnvironmentInfo->getInitialDesign();
+        $this->_restoreInitialInlineTranslation($initialEnvironmentInfo->getData('initial_translate_in_line'));
```
Comment (kkrieger85) [inline app/code/core/Mage/Core/Model/App/Emulation.php]: As far as I read this line it should be

`$this->_restoreInitialInlineTranslation($initialEnvironmentInfo->getData('initial_translate_inline'));`

File: app/code/core/Mage/Core/Model/App/Emulation.php (hunk 2 above)
Comment (sreichel) [reply app/code/core/Mage/Core/Model/App/Emulation.php]: Will be reverted.

File: app/code/core/Mage/Core/Model/App/Emulation.php (hunk 3 above)
Comment (sreichel) [reply app/code/core/Mage/Core/Model/App/Emulation.php]: Will be reverted.

**Comment by sreichel** [comment]
For some reason it looked like "InLine" ... it works since getter/setter have same key, but it will be reverted.

Q: idea was to make phpstan/phpstorm work with Varian_Objects. You cannot use annotations or something else, so the idea was to replace magic methods .... I've already reverted this in another PR and tried something else ... https://github.com/OpenMage/magento-lts/pull/677/commits/088d028850a95af3baabe3e6c91f374d4e24cbfe Maybe this is better?

File: app/code/core/Mage/Core/Helper/Data.php (hunk 1 above)
Comment (sreichel) [reply app/code/core/Mage/Core/Helper/Data.php]: Since we have curly brackets it doesnt matter?

**Pull request closed without merging**